packages/shared/sierra-client/tests/email-verification-notes.test.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  addVerificationNote,
+  deleteNonCurrentVerificationNotes,
+  verifiedEmails,
+} from '../src/email-verification-notes';
+import { VarField, varFieldTags } from '../src/patron';
+
+const notesToVarFields = (...notes: string[]): VarField[] =>
+  notes.map((content) => ({
+    fieldTag: varFieldTags.notes,
+    content,
+  }));
+
+describe('email verification notes', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date('2022-02-22'));
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  describe('addVerificationNote', () => {
+    it('adds a verification note varfield while leaving others alone', () => {
+      const result = addVerificationNote([
+        {
+          fieldTag: varFieldTags.email,
+          content: 'example@example.com',
+        },
+        {
+          fieldTag: varFieldTags.barcode,
+          content: '1234567',
+        },
+      ]);
+
+      expect(result).toHaveLength(3);
+      expect(
+        result.find(({ fieldTag }) => fieldTag === varFieldTags.notes)?.content
+      ).toBe('Auth0: example@example.com verified at 2022-02-22T00:00:00.000Z');
```

Comment:
I'd throw in an assertion that both the original notes are preserved; this test would pass if we put in three copies of our note!